lib/std/crypto/25519/edwards25519.zig

```diff
@@ -4,10 +4,16 @@
 // The MIT license requires this copyright notice to be included in all copies
 // and substantial portions of the software.
 const std = @import("std");
+const crypto = std.crypto;
 const debug = std.debug;
 const fmt = std.fmt;
 const mem = std.mem;
-const Error = std.crypto.Error;
+
+const EncodingError = crypto.errors.EncodingError;
+const IdentityElementError = crypto.errors.IdentityElementError;
+const NonCanonicalError = crypto.errors.NonCanonicalError;
+const NotSquareError = crypto.errors.NotSquareError;
+const WeakPublicKeyError = crypto.errors.WeakPublicKeyError;
 
```

Comment:
`const foo = IdentityElementError || WeakPublicKeyError;` maybe?

Reply:
They have different meanings, so I'm not even sure how such a couple would be named. `InsecureSettingForAPoint`?
But being on the twist may also be an insecure setting for a point, and if we add that third case `InsecureSettingForAPoint`, we'll get a partially used error set in all other cases.

Reply:
Just saw 4 such cases in this file, that's why I suggested to extract it.
If it cannot be named easily then forget about it.

Reply:
No matter how many error sets we have, applications still have to handle individual error cases. 

So, I'd rather avoid trying really hard to cluster errors. Maybe once the whole API is stable, and no new cases will be added. But doing it now is only going to make maintenance more difficult.